### PR TITLE
Widget14. wip

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/120pvb68ay5mvwbdpam13pa328

[download apk](http://filestorage.bluewallet.io:1488/1604137656.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/u2b9bjrxnht5gxy99yyu8rx6w0

[download apk](http://filestorage.bluewallet.io:1488/1604167996.apk)

File: ios/MarketWidget/API.swift [hunk 1]
```diff
@@ -7,9 +7,76 @@
...
+  formatter.groupingSeparator = " "
+  return formatter
+}
 
 class API {
   
+  
+  static func fetchNextBlockFee(completion: @escaping ((MarketData?, Error?) -> Void)) {
+    DispatchQueue.global(qos: .background).async {
+    let client = TCPClient(address: "electrum1.bluewallet.io", port: 50001)
```
Comment (Overtorment) [inline ios/MarketWidget/API.swift:30]: Randomising would be nice if electrum1 is down

File: ios/MarketWidget/API.swift [hunk 2]
```diff
@@ -7,9 +7,76 @@
...
+      switch client.connect(timeout: 1) {
+      case .success:
+        switch client.send(string: send) {
+        case .success:
+          guard let data = client.read(1024*10, timeout: 1) else {
+            client.close()
+            completion(nil, APIError())
+            return
+          }
+          if let response = String(bytes: data, encoding: .utf8), let nextBlockResponse = response.components(separatedBy: #"result":"#).last?.components(separatedBy: ",").first, let nextBlockResponseDouble = Double(nextBlockResponse.trimmingCharacters(in: .whitespacesAndNewlines)) {
```
Comment (Overtorment) [inline ios/MarketWidget/API.swift:42]: I think you just need to look for a newline in response,  no need to make it more complex

**Review by Overtorment — APPROVED** [review]
Lgtm

File: ios/MarketWidget/API.swift (hunk 1 above)
Comment (marcosrdz) [reply ios/MarketWidget/API.swift:30]: I'll be updating it to match the user's electrum settings.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/zhdeyymuf516wy578fw9y3307m

[download apk](http://filestorage.bluewallet.io:1488/1604170020.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/nnxgm5kykebzc7u20ucyugqcw4

[download apk](http://filestorage.bluewallet.io:1488/1604172380.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/m4ed4eenxdcb12vbzau64uxqvg

[download apk](http://filestorage.bluewallet.io:1488/1604182311.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/q0dp3ctr2ekfkh9kmg33ub3mx0

[download apk](http://filestorage.bluewallet.io:1488/1604192402.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/7curvnkk6ez8fqhjtwdd44hnqg

[download apk](http://filestorage.bluewallet.io:1488/1604201925.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/p72bk8n3w2w9ve6dmjbrvff0ar

[download apk](http://filestorage.bluewallet.io:1488/1604210585.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/a5adfb4wukdbwg5hvywkxjb8yw

[download apk](http://filestorage.bluewallet.io:1488/1604217180.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/rqm0nutgz2qu214ahnb931wfp8

[download apk](http://filestorage.bluewallet.io:1488/1604296803.apk)

**Pull request merged**